src/neo-vm/ExecutionEngine.cs

```diff
@@ -70,6 +71,18 @@ public ExecutionEngine()
 
         protected virtual void ContextUnloaded(ExecutionContext context)
         {
+            if (InvocationStack.Count == 0 || context.StaticFields != InvocationStack.Peek().StaticFields)
+            {
+                context.StaticFields?.ClearReferences();
+            }
+            context.LocalVariables?.ClearReferences();
+            context.Arguments?.ClearReferences();
+        }
+
+        internal ExecutionContext PopExecutionContext()
+        {
+            CurrentContext = InvocationStack.Pop();
```

Comment:
You should not set `CurrentContext` to a popped item.

Reply:
Done